torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -120,9 +128,97 @@ TORCH_CUDA_API Val* ceilDiv(Val* v1, Val* v2) {
 }
 
 TORCH_CUDA_API Val* andOp(Val* v1, Val* v2) {
+  TORCH_CHECK(
+      v1->getDataType().value() == DataType::Bool,
+      "Input1 should be of type bool, not ",
+      v1->getDataType().value());
+  TORCH_CHECK(
+      v2->getDataType().value() == DataType::Bool,
+      "Input2 should be of type bool, not ",
+      v2->getDataType().value());
   return binaryOp(BinaryOpType::And, v1, v2);
 }
 
+TORCH_CUDA_API Val* add_alpha(Val* v1, Val* v2, Val* s) {
```

Comment:
NITPICK: this API is not used in parser.
IIUC, the problem is that this API hides `intrm`, which will need to be manually binded to thread/block depends on how loops are nested.

(I don't see this as a problem, this is not a super complicated utility function and parser have similar logic that handles this)

Reply:
same thing for sub_alpha

Reply:
Actually we should be able to use this interface...
https://github.com/csarofeen/pytorch/blob/50e9c95eb8182cf565c404307c2480c4e8b55192/torch/csrc/jit/codegen/cuda/parser.cpp#L103-L124
These for loops are actually not right, there should only be a single for loop, and it should be based on:
`for (auto val : fusion_->vals())`
This loop will grab every intermediate tensor view in the fusion and set bindings correctly. Then you can use this interface as we're using others.